File: app10_csharp_react/backend/src/SharpGuard.Api/Entities/Framework.cs
namespace SharpGuard.Api.Entities;

public class Framework
{
    public Guid Id { get; set; }
    public required string Code { get; set; }
    public required string Name { get; set; }
    public required string Version { get; set; }
    public string? Description { get; set; }
    public string? ReferenceUrl { get; set; }

    public List<Threat> Threats { get; set; } = [];
}

File: app11_swift_ios/SwiftGuardData/Sources/SwiftGuardData/Models/Framework.swift
import Foundation
import SwiftData

@Model
public final class Framework {
    @Attribute(.unique) public var code: String   // "OWASP_WEB", "OWASP_LLM", "MITRE_ATLAS", ...
    public var name: String
    public var version: String
    public var frameworkDescription: String
    public var referenceUrl: String
    @Relationship(deleteRule: .cascade, inverse: \Threat.framework) public var threats: [Threat] = []

    public init(code: String, name: String, version: String, frameworkDescription: String, referenceUrl: String) {
        self.code = code
        self.name = name
        self.version = version
        self.frameworkDescription = frameworkDescription
        self.referenceUrl = referenceUrl
    }
}
